app/src/main/java/org/oppia/android/app/options/ReadingTextSizeActivityPresenter.kt

```diff
@@ -26,6 +28,13 @@ class ReadingTextSizeActivityPresenter @Inject constructor(
     this.fontSize = fontSize
   }
 
+  private fun setToolbar() {
+    val readingTextSizeToolbar: Toolbar = activity.findViewById(R.id.reading_text_size_toolbar)
+    readingTextSizeToolbar.setNavigationOnClickListener {
+      activity.onBackPressed()
+    }
+  }
```

Comment:
You forgot to include the selectedReadingTextSize in the result for this Activity, compare it with the fragment to understand the difference

Reply:
@ARJUPTA  what I thought was, that functions onBackPressed is already overridden in ReadingTextSizeActivity.kt file which will handle everything.

Reply:
`ReadingTextSizeActivity` will be able to handle backPress but we also need to set the updated test size data in the activity result (which can be read by previous activity).